File: solutions/go/lasagna-master/2/lasagna_master.go
package lasagna

func PreparationTime(layers []string, prepTime int) int {
	if prepTime == 0 {
		return len(layers) * 2
	} else {
		return len(layers) * prepTime
	}
}

func Quantities(layers []string) (noodles int, sauce float64) {
	for _, l := range layers {
		switch(l) {
		case "noodles":
			noodles += 50
		case "sauce":
			sauce += 0.2
		}
	}
	return noodles, sauce
}

func AddSecretIngredient(friendsList []string, myList []string) {
	myList[len(myList) - 1] = friendsList[len(friendsList) - 1]
}

func ScaleRecipe(portions []float64, numberOfPortions int) []float64 {
	desiredQuantity := []float64{}

	for i := 0; i < len(portions); i++ {
		desiredQuantity = append(desiredQuantity, (portions[i] * float64(numberOfPortions)) / 2)
	}

	return desiredQuantity
}

File: solutions/javascript/lasagna-master/1/lasagna-master.js
/// <reference path="./global.d.ts" />
// @ts-check

/**
 * Implement the functions needed to solve the exercise here.
 * Do not forget to export them so they are available for the
 * tests. Here an example of the syntax as reminder:
 *
 * export function yourFunction(...) {
 *   ...
 * }
 */
export function cookingStatus(remainingTime) {
  if (remainingTime === undefined) {
    return 'You forgot to set the timer.';
  }
  if (remainingTime === 0) {
    return 'Lasagna is done.';
  }
  return 'Not done, please wait.';
}

export function preparationTime(layers, prepTime = 2) {
  return layers.length * prepTime;
}

export function quantities(layers) {
  const gramsOfNoodles = layers.filter(layer => layer === 'noodles').length * 50;
  const litersOfSauce = layers.filter(layer => layer === 'sauce').length * 0.2;

  return {
    noodles: gramsOfNoodles,
    sauce: litersOfSauce,
  };
}

export function addSecretIngredient(friendList, myList) {
  const specialIngredient = friendList[friendList.length - 1];
  myList.push(specialIngredient);
}

export function scaleRecipe(recipe, portionSize) {
  let result = {};
  for (const [key, value] of Object.entries(recipe)) {
    result[key] = (value * portionSize) / 2;
  }
  return result;
}
